controllers/front/SmailyCustomerCron.php

```diff
@@ -24,83 +24,60 @@
 
 class SmailyforprestashopSmailyCustomerCronModuleFrontController extends ModuleFrontController
 {
+    /**
+     * Limit unsubscribers request batch size.
+     */
+    const UNSUBSCRIBERS_BATCH_LIMIT = 1000;
+
+    /**
+     * Limit subscribers query batch size.
+     */
+    const SUBSCRIBERS_BATCH_LIMIT = 1000;
+
     public function init()
     {
         parent::init();
         header('Content-Type: text/plain');
-        if (Tools::getValue('token') == Configuration::get('SMAILY_CUSTOMER_CRON_TOKEN')) {
-            $this->syncContacts();
-            die();
-        } else {
+
+        if (Tools::getValue('token') != Configuration::get('SMAILY_CUSTOMER_CRON_TOKEN')) {
             die($this->l('Access denied! '));
         }
+    
+        if ((int) Configuration::get('SMAILY_ENABLE_CRON') !== 1) {
+            die($this->l('User synchronization disabled!'));
+        }
+
+        if ($this->syncContacts()) {
+            die($this->l('User synchronization done!'));
+        } else {
+            die($this->l('User synchronization failed!'));
```

Comment:
`die` will exit with status code `0`. Doesn't PrestaShop have any helper methods for terminating scripts (like Wordpress has `wp_die`)?

Reply:
Yes, I think they use die() functionality everywhere. Is there any point in returning a status code if you are not running cron from file? I think Prestashop cron can be only run if you visit the URL and the message back is sent in plain text response.

Reply:
Also, I think this translation is unnecessary here. I don't think the store language context is even initiated and if you want to build some kind of logic on cURL call response then the translations will only mess things up.

Reply:
Yes, if PrestaShop URL is being called by CRON job. For example developer might have set up something like this:

```
0 2 * * * /path/to/monitoring-wrapper.sh curl http://localhost/prestashop/index.php?controller=SyncController&token=... >> /var/log/cron.log 2&>1
```

If cURL call to our controller returns a response that forces cURL to exit with non-zero exit code then `monitoring-wrapper.sh` could capture it and forward the information (for example) to an email address.

But if PrestaShop does not have any best practices defined for such use case or other CRON controllers only use `die` then I guess `die` works fine.

Reply:
Yes, you are right. Translations aren't necessary here. The response is meant for a developer.

Reply:
I could do an
```
echo 'Access denied!';
die(1);
```
Would that be a better approach?

Reply:
Yes.

Reply:
Changed to `echo` and `die` solution and removed translations. https://github.com/sendsmaily/smaily-prestashop-module/pull/10/commits/55119959176cc9f057613e611db2f9c02593b5b0